KEYCLOAK-6618 Update German translations

ferdinandhuebner opened the pull request:
- Add missing translations for OTP authenticator settings and update outdated translations for OTP authenticator
- Fix minor issue for the username property (plural -> singular)
- Add missing translations

Commit:
KEYCLOAK-6618 Update German translations

Add missing translations for OTP authenticator settings and update
outdated translations for OTP authenticator

Fix minor issue for the username property (plural -> singular)

Add missing translations

pilhuhn commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 identityProviderRedirectErrorMessage=Fehler bei der Weiterleitung zum Identity Provider.
 identityProviderRemovedMessage=Identity Provider erfolgreich entfernt.
+identityProviderAlreadyLinkedMessage=Die f\u00F6derierte Identit\u00E4t von {0} ist bereits einem anderen Benutzer zugewiesen.
+staleCodeAccountMessage=Die Seite ist abgelaufen. Bitte versuchen Sie es noch einmal.
```
"Seite abgelaufen" sounds a bit strange.

thomasdarimont replied:
Vielleicht "Diese Seite ist nicht mehr gültig, bitte versuchen Sie es noch einmal"

thomasdarimont replied:
Vielleicht "Diese Seite ist nicht mehr gültig, bitte versuchen Sie es noch einmal"

ferdinandhuebner replied:
I like Thomas' suggestion and am going to pick that.

pilhuhn reviewed (commented):
In total sounds good. I did not check with the English original messages to verify the content really makes sense, but at least the translations match the keys

danielkocot commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
+totpUnableToScan=Sie K\u00F6nnen den Barcode nicht scannen?
+totpScanBarcode=Barcode scannen?
+
+totp.totp=Zeitbasiert (time-based)
```
zeitbasiert

ferdinandhuebner replied:
Apart from the capitalization, would you suggest that I omit the "(time-based)" part? 

I was uncertain whether to include it or not. If we only display the German word, users might get confused, especially if their authenticator application is not or not properly translated.

danielkocot replied:
You shouldn't omit the "(time-based)" part.

danielkocot commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
+totpScanBarcode=Barcode scannen?
+
+totp.totp=Zeitbasiert (time-based)
+totp.hotp=Z\u00E4hlerbasiert (counter-based)
```
z\u00E4hlerbasiert

danielkocot commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
 missingLastNameMessage=Bitte geben Sie einen Nachnamen ein.
+missingEmailMessage=Bitte geben Sie eine E-Mail Adresse ein.
 missingPasswordMessage=Bitte geben Sie ein Passwort ein.
 notMatchPasswordMessage=Passw\u00F6rter sind nicht identisch.
```
Die Passw\u00F6rter sind nicht identisch.

danielkocot commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
+staleCodeAccountMessage=Die Seite ist abgelaufen. Bitte versuchen Sie es noch einmal.
+consentDenied=Einverst\u00E4ndnis verweigert.
 
 accountDisabledMessage=Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
```
Ihr Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.

danielkocot commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 
 accountDisabledMessage=Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
 
 accountTemporarilyDisabledMessage=Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
```
Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.

danielkocot reviewed (changes requested):
Corrected translations in the comments.

ferdinandhuebner commented:
Thanks for the feedback. 
Would you prefer that I use a separate commit to address your comments and rebase and squash after the review is done, or squash right away?

Commit:
KEYCLOAK-6618: Include review feedback into German translations

ferdinandhuebner commented:
I revised the problematic translations. 

If the changes are OK for you, I can rebase and squash so this PR can be merged.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
 
 configureAuthenticators=Authenticatoren konfigurieren
 mobile=Mobile
```
Mobil

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
+totpManualStep2=\u00D6ffnen Sie die Applikation und geben Sie den folgenden Schl\u00FCssel ein.
+totpManualStep3=Verwenden Sie die folgenden Konfigurationswerte, falls Sie diese f\u00FCr die Applikation anpassen k\u00F6nnen:
+totpUnableToScan=Sie k\u00F6nnen den Barcode nicht scannen?
+totpScanBarcode=Barcode scannen?
```
Is it a question?

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
+readOnlyUsernameMessage=Sie k\u00F6nnen Ihren Benutzernamen nicht \u00E4ndern, da er schreibgesch\u00FCtzt ist.
+readOnlyPasswordMessage=Sie k\u00F6nnen Ihr Passwort nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
 
 successTotpMessage=Mobile Authentifizierung eingerichtet.
```
Mehrfachauthentifizierung erfolgreich konfiguriert.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
+readOnlyPasswordMessage=Sie k\u00F6nnen Ihr Passwort nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
 
 successTotpMessage=Mobile Authentifizierung eingerichtet.
 successTotpRemovedMessage=Mobile Authentifizierung entfernt.
```
Mehrfachauthentifizierung erfolgreich entfernt.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
+accountTemporarilyDisabledMessage=Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
 invalidPasswordMinLengthMessage=Ung\u00FCltiges Passwort\: Minimall\u00E4nge {0}.
-invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
```
Ung\u00FCltiges Passwort\: Es muss mindestens {0} Kleinbuchstaben beinhalten.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 
-accountTemporarilyDisabledMessage=Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
+accountTemporarilyDisabledMessage=Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
 invalidPasswordMinLengthMessage=Ung\u00FCltiges Passwort\: Minimall\u00E4nge {0}.
```
Ung\u00FCltiges Passwort\: Es muss mindestens {0} Zeichen lang sein.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 invalidPasswordMinLengthMessage=Ung\u00FCltiges Passwort\: Minimall\u00E4nge {0}.
-invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
+invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
```
Ung\u00FCltiges Passwort\: Es muss mindestens {0} Zahl(en) beinhalten.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
-invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
+invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinUpperCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Grossbuchstaben beinhalten.
```
Ung\u00FCltiges Passwort\: Es muss mindestens {0} Gro\u00DFbuchstaben beinhalten.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
+invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinUpperCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Grossbuchstaben beinhalten.
 invalidPasswordMinSpecialCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Spezialzeichen beinhalten.
```
Ung\u00FCltiges Passwort\: Es muss mindestens {0} Sonderzeichen beinhalten.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 invalidPasswordMinUpperCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Grossbuchstaben beinhalten.
 invalidPasswordMinSpecialCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Spezialzeichen beinhalten.
-invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie Benutzername.
+invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie der Benutzername.
```
Ung\u00FCltiges Passwort\: Es darf nicht gleich sein wie der Benutzername.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 invalidPasswordMinSpecialCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Spezialzeichen beinhalten.
-invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie Benutzername.
+invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie der Benutzername.
 invalidPasswordRegexPatternMessage=Ung\u00FCltiges Passwort\: nicht Regex-Muster (n) entsprechen.
```
Ung\u00FCltiges Passwort\: Es entspricht nicht dem Regex-Muster.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 invalidPasswordRegexPatternMessage=Ung\u00FCltiges Passwort\: nicht Regex-Muster (n) entsprechen.
-invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
\ No newline at end of file
+invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
```
Ung\u00FCltiges Passwort: Es darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.

muehlburger commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
-invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
\ No newline at end of file
+invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
+invalidPasswordBlacklistedMessage=Ung\u00FCltiges Passwort: Das Passwort ist in der Schwarzen Liste enthalten.
```
Ung\u00FCltiges Passwort: Das Passwort steht auf der Blocklist (schwarzen Liste).

muehlburger reviewed (changes requested):
Hi, I reviewed the changes and suggest some minor updates.

ferdinandhuebner commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -24,7 +24,7 @@ familyName=Nachname
 password=Passwort
 passwordConfirm=Passwortbest\u00E4tigung
```
That label is used when the password is changed. The english value of the label is "Confirmation".

I'm going to change it to "Passwort best\u00E4tigen". Other sites, e.g. Google, use that phrase and it is used in the login theme as well.

muehlburger replied:
That sounds good.

ferdinandhuebner commented on themes/src/main/resources-community/theme/base/account/messages/messages_de.properties:
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
 
 configureAuthenticators=Authenticatoren konfigurieren
```
I'm happy to change that. However, I would suggest rewording "Authentifikatoren" and "Authenticatoren" to "Mehrfachauthentifizierung" throughout the entire theme. 

It would be strange if we call it "Mehrfachauthentifizierung" when displaying the result of activating or deactivating it, but call it "Authentifikatoren" in the menu and page title. 

TOTP labels are used in the login theme as well. However, it is called "Mobile Authentifizierung" in that theme. 
For the sake of consistency, the same phrase should be used in both themes. Would you prefer "Mobile Authentifizierung" or "Mehrfachauthentifizierung"?

I'll go through the translations of the other themes, because they seem incomplete. Would you prefer that I open another pull-request for hat or update this one?

muehlburger replied:
Definitely "Mehrfachauthentifizierung". It refers to to multi-factor-authentication as well as two-factor authentication.

What does "mobile authentication" mean exactly? I understand it as the possibility to configure two-factor authentication, which uses for example Google Authenticator or FreeOTP. If this was the case it should not be called "Mobile Authentication" but "Mehrfachauthentifizierung".

ferdinandhuebner replied:
Yes, the term is always used in that context. I reworded it to "Mehrfachauthentifizierung" troughout the account theme and the login theme.

Commit:
KEYCLOAK-6618: Reword translation for multi-factor authentication and fix minor translation issues

Commit:
KEYCLOAK-6618: Update German translation for the login theme

Message bundle keys have been reordered to be in sync with the english
version to improve scanning through the message bundles side-by-side.

The updated German translations from the account theme were applied to
the login theme as well (where applicable).

ferdinandhuebner commented:
As discussed with @muehlburger, I reworded the German translation for multi-factor authentication to "Mehrfachauthentifizierung" instead of "Mobile Authentifizierung".

Commit 0a4059804713b54974c2fae94706d07a35578258 applies the changes we discussed for the account theme to the login theme which was incomplete.
